### PR TITLE
seqan3: update to 3.0.3

### PR DESCRIPTION
**Pull request by eseiler**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.15
macports-gcc-10

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: science/seqan3/Portfile [hunk 1]
```diff
@@ -28,6 +28,9 @@ github.tarball_from   releases
 distname              ${github.project}-${github.version}-Source
 use_xz                yes
 
+depends_run           port:bzip2 \
```
Comment (reneeotten) [inline science/seqan3/Portfile:31]: I suspect these dependencies are also needed at configure/build time. If that’s indeed the case you should use “depends_lib-append” (using append is preferred anyway as not to accidentally overwrite dependencies that might be added by portgroups).

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: science/seqan3/Portfile (hunk 1 above)
Comment (eseiler) [reply science/seqan3/Portfile]: Thank you for the explanation, I changed it accordingly.

**Review by reneeotten — APPROVED** [review]

**Comment by reneeotten** [comment]
@rrahn are you okay with these changes? Also since you didn't add `openmaintainer` to this port it would be good that you'll get notified when a PR is opened for `seqan3`; for that to work one would need to add your GitHub handle to the `maintainer` field - are you okay with doing that?

**Review by rrahn — APPROVED** [review]
lgtm. Thank you

**Comment by rrahn** [comment]
> @rrahn are you okay with these changes? Also since you didn't add `openmaintainer` to this port it would be good that you'll get notified when a PR is opened for `seqan3`; for that to work one would need to add your GitHub handle to the `maintainer` field - are you okay with doing that?

Hi @reneeotten yes, I am ok with that. Thank you very much

**Comment by reneeotten** [comment]
@rrahn thank you for the quick response! Is adding `openmaintainer` to the Portfile also okay with you - in that case others can do (minor) updates to the Portfile without your explicit permission?

**Comment by rrahn** [comment]
Sounds great. Many thanks.

**Pull request merged**